File: code/first_class_citizens_input.py
def multiply_by_2(x):  # first-order function
    return x * 2


def apply_to_five(f):  # higher-order function, takes a function as its input
    return f(5)


print(f"Value of apply_to_five(multiply_by_2): {apply_to_five(multiply_by_2)}.")

File: code/first_class_citizens_input.hs
multiply_by_2(x) = x * 2    -- first-order function
apply_to_five(f) = f(5)     -- higher-order function, takes a function as its input

msg = "Value of apply_to_five(multiply_by_2): " ++ show(apply_to_five(multiply_by_2)) ++ "."
main = putStrLn(msg)
